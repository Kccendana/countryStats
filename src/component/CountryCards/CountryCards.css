ul.countryList {
  display: grid;
  grid-template-columns: 1fr 1fr;
  color: white;
}

.cards {
  width: auto;
  position: relative;
}

.info {
  display: flex;
  align-items: flex-end;
  flex-direction: column;
  padding-right: 1em;
}

ul.countryList li {
  background-color: #db4681;
  padding-bottom: 1.25em;
}

ul.countryList li a {
  color: white;
  text-decoration: none;
  cursor: pointer;
}

.countryList li:nth-child(4n-1),
.countryList li:nth-child(4n-2) {
  background-color: #c04778;
}

img.flag {
  max-width: 100%;
  height: auto;
}

.info h2 {
  font-size: 1.05em;
  margin-top: 0.5em;
  text-align: right;
  font-weight: 700;
  text-transform: uppercase;
}

.img-con {
  position: relative;
  overflow: hidden;
  margin: 1.25em;
}

.button {
  border: 2px solid white;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background-color: #e83779;
  color: white;
  font-size: 0.75em;
  position: absolute;
  top: 10px;
  right: 10px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
}

.subtitle {
  color: white;
  font-weight: 500;
  font-size: 0.85em;
  padding: 0.25em 1em;
  background-color: #cf4378;
  text-align: left;
  text-transform: uppercase;
}

@media only screen and (min-width: 49em) {
  ul.countryList {
    grid-template-columns: 1fr 1fr 1fr;
  }

  ul.countryList li:nth-child(even) {
    background-color: #c04778;
  }

  ul.countryList li:nth-child(odd) {
    background-color: #db4681;
    padding-bottom: 1.25em;
  }
}
