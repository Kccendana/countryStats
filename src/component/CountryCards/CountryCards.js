import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowRightLong } from '@fortawesome/free-solid-svg-icons';
import { Link } from 'react-router-dom';
import { countriesByRegion, getAllCountries } from '../../redux/countrySlice';
import './CountryCards.css';

function formatPopulation(population) {
  // Use regular expression to format population with commas
  return population.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
}

function CountryCards() {
  const { countries, region } = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  useEffect(() => {
    if (region) {
      dispatch(countriesByRegion(region));
    } else {
      dispatch(getAllCountries());
    }
  }, [dispatch, region]);

  return (
    <div>
      <p className="subtitle">{region ? `Countries in ${region.charAt(0).toUpperCase() + region.slice(1)}` : 'All countries'}</p>
      <ul className="countryList">
        {countries.map((country) => (
          <li key={country.name.common}>
            <Link to={`/${country.cca2}`}>
              <div className="cards">
                <div className="img-con">
                  <img className="flag" src={country.flags.png} alt={country.name.common} />
                  <div className="overlay" />
                </div>
                <div className="info">
                  <h2>{country.name.common}</h2>
                  <span>{formatPopulation(country.population)}</span>
                </div>
                <div className="button"><FontAwesomeIcon icon={faArrowRightLong} /></div>
              </div>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default CountryCards;
