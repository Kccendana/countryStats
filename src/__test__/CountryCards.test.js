import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CountryCards from '../component/CountryCards/CountryCards';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('renders CountryCards component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <CountryCards />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
