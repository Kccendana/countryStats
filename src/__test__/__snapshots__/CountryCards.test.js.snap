// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CountryCards component matches the snapshot 1`] = `
<div>
  <div>
    <p
      class="subtitle"
    >
      All countries
    </p>
    <ul
      class="countryList"
    />
  </div>
</div>
`;
